pkg/apis/openebs.io/v1alpha1/types.go

```diff
@@ -142,3 +150,50 @@ type Task struct {
 	// Kind is the kind corresponding to the task's actual content
 	Kind string `json:"kind"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=cstorcrd
+
+// CstorCRD describes a CstorCRD.
+type CstorCrd struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata",omitempty`
+
+	Spec CstorCrdSpec `json:"spec"`
+}
+
+// CstorCRDSpec is the spec for a StoragePoolClaimSpec resource
+type CstorCrdSpec struct {
+	Zpool Zpool `json:"zpool"`
+}
+type Zpool struct {
+	Poolname      string `json:"poolname"`
+	Provisiontype string `json:"provisiontype"` //thick,thin
+	Cachefile     string `json:"cachefile"`
+	Pooltype      string `json:"pooltype"` //mirror, striped, raid
+	DiskPath      string `json:"diskPath"`
+	Zfs           Zfs    `json:"zfs"`
```

Comment:
what doe this value mean?

Reply:
Zfs is another struct for running zfs commands. Here I am using Zfs struct inside Zpool since a pool may have any number of volumes. We are now thinking of having separate crds for zfs and zpool, since pool is not going to be created often.